homeassistant/components/climate/nuheat.py

```diff
@@ -0,0 +1,212 @@
+"""
+Support for NuHeat thermostats.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/climate.nuheat/
+"""
+import logging
+from datetime import timedelta
+
+from homeassistant.components.climate import (
+    ClimateDevice,
+    STATE_HEAT,
+    STATE_IDLE)
+from homeassistant.components.nuheat import DATA_NUHEAT
+from homeassistant.const import (
+    ATTR_TEMPERATURE,
+    TEMP_CELSIUS,
+    TEMP_FAHRENHEIT)
+from homeassistant.util import Throttle
+
+DEPENDENCIES = ["nuheat"]
+
+_LOGGER = logging.getLogger(__name__)
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=5)
+
+MODE_AUTO = "auto"  # Run device schedule
+MODE_AWAY = "away"
+MODE_HOLD_TEMPERATURE = "temperature"
+MODE_TEMPORARY_HOLD = "temporary_temperature"
+
+OPERATION_LIST = [STATE_HEAT, STATE_IDLE]
+
+SCHEDULE_HOLD = 3
+SCHEDULE_RUN = 1
+SCHEDULE_TEMPORARY_HOLD = 2
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the NuHeat thermostat(s)."""
+    if discovery_info is None:
+        return
+
+    _LOGGER.info("Loading NuHeat thermostat climate component")
+    temperature_unit = hass.config.units.temperature_unit
+    api, serial_numbers = hass.data[DATA_NUHEAT]
+    thermostats = [
+        NuHeatThermostat(api, serial_number, temperature_unit)
+        for serial_number in serial_numbers
+    ]
+    add_devices(thermostats, True)
+
+
+class NuHeatThermostat(ClimateDevice):
+    """Representation of a NuHeat Thermostat."""
+    def __init__(self, api, serial_number, temperature_unit):
+        self._thermostat = api.get_thermostat(serial_number)
+        self._temperature_unit = temperature_unit
+        self._force_update = False
+
+    @property
+    def name(self):
+        """Return the name of the thermostat."""
+        return self._thermostat.room
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        if self._temperature_unit == "C":
+            return TEMP_CELSIUS
+
+        return TEMP_FAHRENHEIT
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        if self._temperature_unit == "C":
+            return self._thermostat.celsius
+
+        return self._thermostat.fahrenheit
+
+    @property
+    def current_operation(self):
```

Comment:
You define an "auto" mode above, yet it seems never to be returned from this?

Reply:
The `auto` mode you're referring to was a type of `hold_mode`, not a `current_operation`

After reading this: https://home-assistant.io/components/climate/#service-climateset_hold_mode

I determined that, in NuHeat, there are effectively 4 different kinds of hold modes.
* Run schedule programmed into thermostat (`auto`) - although after re-reading the documentation, maybe this should be `home`
* Indefinite hold (`temperature`)
* temporary hold (`temporary_temperature`) - holds the temperature until the next scheduled change programmed into the thermostat. after reading the 
* away (`away`) - holds a given minimum temperature until away_mode is turned off.

Reply:
So, if the thermostat is in `auto` mode (or `home`), it does not allow manual changes in temperature? Or how does `auto` differ from `temporary_temperature`? Do we need this `auto` mode then if the only advantage over `temporary_temperature` is that it prevents values from being entered?

One more thing: please call it `auto`, not `home`. Calling it `home` just adds one more choice for the state. ;-)

Reply:
When the thermostat is running its schedule and you're in the bathroom and decide that the current floor temp is not where you want it, you can adjust it up or down (via up/down buttons on the hardware)... when doing this it will stay at that temperature until the next scheduled target temperature occurs.

i.e. given this simple Thermostat schedule:
- 12am-6am: 72 degrees
- 6am-9am: 80 degrees
- 9am-5pm: 72 degrees
- 5pm-12am: 80 degrees

When this schedule is running it is in `auto` / `home` mode.

If I'm home sick and take a late shower at noon, my floor will be 72 degrees... but I'm cold, so I bump it up to 85 degrees. The thermostat will temporarily hold the floor at 85 degrees until the next scheduled event at 5pm... where it will drop to 80 and be back in `auto` mode.

Indefinite hold, temporary hold, and run schedule (`auto`) are simply modes supported by the thermostat and by the python library in use.

Reply:
Okay, I think I get it. In your example, the `hold mode` would go to `temporary_temperature`, right? And if you switched it back to `auto` / `home` (from home assistant), the temperature would go back to 72 degrees?

Reply:
yep, that's exactly how it works.